python/Ganga/__init__.py

```diff
@@ -727,6 +727,15 @@ def filefilter(fn):
                         'defaultProtocol': defaultMassStorageProto},
                        massStorageFileExt)
 
+sharedFileBackendPost = {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client', 'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}
+
+output_config.addOption('SharedFile',
+                       {'fileExtensions': [''],
+                        'backendPostprocess': sharedFileBackendPost,
+                        'uploadOptions': {'path': '/tmp/sharedStorage', 'cp_cmd': 'cp', 'ls_cmd': 'ls', 'mkdir_cmd': 'mkdir -p'},
```

Comment:
But surely if a user tries to use the class (even accidently) and hasn't configured it correctly then it **should** throw an error? It would only need to be checked when the class actually does anything (or is instantiated maybe).

Btw, the `mkdir` command shouldn't have the `-p` option as this will **always** create all directories irrespective of the code in `MassStorageFile`.